pkgs/development/interpreters/gnudatalanguage/default.nix

```diff
@@ -0,0 +1,156 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, wrapGAppsHook
+, readline
+, ncurses
+, zlib
+, gsl
+, openmp
+, graphicsmagick
+, netcdf
+, hdf4
+, hdf5
+, fftw
+, fftwFloat
+, fftwLongDouble
+, proj
+, shapelib
+, expat
+, udunits
+, eigen
+, pslib
+, eccodes
+, glpk
+, libpng
+, plplot
+, libtiff
+, libgeotiff
+, libjpeg
+, enableLibtirpc ? stdenv.isLinux, libtirpc
+, pythonInterpreter ? python3, python3
+, enableMPI ? (stdenv.isLinux || stdenv.isDarwin)
+# Choose mpich because of it builds on aarch and darwin
+, libraryMPI ? mpich
+, mpich
+# Unfree optional dependency
+# for hdf4 and hdf5
+, enableSzip ? false, szip
+# wxWidgets is preferred over X11 for this project
+# but we only have it on Linux and Darwin in Nixpkgs
+, enableWX ? (stdenv.isLinux || stdenv.isDarwin)
+, wxGTK31, wxmac
+# X11: OFF by default for platform consistency
+# Use X where WX is not available
+, enableXWin ? (!stdenv.isLinux && !stdenv.isDarwin)
+, xorg
+}:
+
+let
+  wxWidgets = (if stdenv.isDarwin then wxmac else wxGTK31);
+  enablePlplotDrivers = enableWX || enableXWin;
+  hdf4-custom = (hdf4.override (
+    {
+      uselibtirpc = enableLibtirpc;
+    } // (lib.optionalAttrs enableSzip {
+      inherit szip;
+    })
+  ));
+  hdf5-custom = (hdf5.override (
+    (lib.optionalAttrs enableMPI {
+      mpiSupport = true;
+      mpi = libraryMPI;
+    }) // (lib.optionalAttrs enableSzip {
+      inherit szip;
+    })
```

Comment:
I think it would be better to:

- Edit the expressions for these hdf packages so that you will be able to `inherit enableSzip enableMpi`.
- Add a passthru to these hdf packages so you'd be able to access the `mpi` and `szip` dependencies used there, here.